File: array_and_hashing/cpp/longestCommonPrefix.cpp
#include <vector>
#include <string>
using namespace std;

string longestCommonPrefix(vector<string>& strs) {
    string result = "";
    for(int i = 0; i < strs[0].size(); i++){
        for(int j = 1; j < strs.size(); j++){
            if(strs[0][i] != strs[j][i]){
                return result;
            }
        }
        result += strs[0][i];
    }
    return result;
}

File: array_and_hashing/swift/longestCommonPrefix.swift
func longestCommonPrefix(_ strs: [String]) -> String {
    var result : String = ""
    for i in 0..<strs[0].count{
        var currLetter = Array(strs[0])[i]
        for j in 1..<strs.count{
            if(i == strs[j].count || currLetter != Array(strs[j])[i]){
                return result
            }
        }
        result += String(currLetter)
    }
    return result
}